src/server/req_quejob.c

```diff
@@ -241,6 +243,74 @@ validate_perm_res_in_select(char *val, int val_exist)
 #define SET_RESC_SELECT	1
 #define SET_RESC_PLACE	2
 
+#ifndef PBS_MOM
+/**
+ * @brief	Calculate the index of the current server
+ *
+ * @param	void
+ *
+ * @return	int
+ * @retval	index of the server
+ * @retval	-1 if couldn't be determined
+ */
+static int
+get_server_index(void)
+{
+	int i;
+
+	for (i = 0; i < get_num_servers(); i++) {
+		if (pbs_conf.psi[i].port == pbs_server_port_dis && is_same_host(pbs_conf.psi[i].name, server_host))
+			return i;
+	}
+
+	return -1;
+}
+
+/**
+ * @brief	Generate and fill jobid of the new job
+ * 			Note: Consider directly modifying get_next_svr_sequence_id() if reservations
+ * 			also get sharded for multi-server
+ *
+ * @param[out]	jidbuf - buffer to fill job id in
+ * @param[in]	clusterid - cluster name (PBS_SERVER)
+ * @param[in]	jtype - job type (0 for normal job, 1 for job array)
+ *
+ * @return	int
+ * @retval	0 for Success
+ * @retval	1 for Failure
+ */
+static int
+generate_jobid(char *jidbuf, char *clusterid, int jtype)
```

Comment:
since this is intended to be a switchable function, is it better to move it to a separate/new code file so that it can be easily patched or moved to a shared library in future? Or, should we make this a shared library to start with?

Reply:
I hesitate doing this preemptively without knowing what the future holds. Do we want to design for different sites wanting different jobid generation schemes now? Maybe it'll get designed as a configuration (qmgr/pbs.conf) switch instead of an installation time decision (via shared libraries)? So, I feel like we should wait for further feedback. What do you think?

Reply:
I agree - nevertheless just wanted to get u started to think about this :) Let's leave it as is for now